forte/data/readers/classification_reader.py

```diff
@@ -0,0 +1,177 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import csv
+import importlib
+from typing import Iterator, Tuple
+from collections import OrderedDict
+from forte.common import Resources
+from forte.common.configuration import Config
+from forte.data.data_pack import DataPack
+from forte.data.base_reader import PackReader
+from ft.onto.base_ontology import *
+
+
+__all__ = ["ClassificationDatasetReader"]
+
+
+class ClassificationDatasetReader(PackReader):
+    r"""
+    A generic dataset reader class for classification datasets.
```

Comment:
Let's explain in details the scope of this reader, how to use it, and provide an example in this class docstring.